pkg/proxy/downstream.go

```diff
@@ -1361,7 +1365,7 @@ func (s *downStream) doRetry() {
 	// no reuse buffer
 	atomic.StoreUint32(&s.reuseBuffer, 0)
 
-	pool, err := s.initializeUpstreamConnectionPool(s)
+	host, pool, err := s.initializeUpstreamConnectionPool(s)
```

Comment:
should reset requestInfo， meybe we can set reqeustinfo in the func of initializeUpstreamConnectionPool()